File: components/hcfGPU.java
package components; // ← Fará parte do pacote "components", junto com outras peças de computador.
// ↓ Esta classe é herdada da GPU, para placas de vídeo Radeon com Hybrid Crossfire
public class hcfGPU extends GPU {
 private String igpu; // ← Placa de vídeo integrada compatível para Hybrid Crossfire
 private boolean hcf; // ← Tecnologia desligada por padrão
 private String valor;
 public hcfGPU (String nome, Integer gb, String tipo, String conn, String igpu) {
  super(nome, gb, tipo, conn);
  if (igpu == null || igpu.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  this.igpu=igpu;
  this.hcf=false;
  this.valor = (hcf) ? "Ativado" : "Desativado";
 }
 @Override
 public String toString() {
  return "\n::::: TECHPOWERUP GPU DATABASE :::::\nNome: "+nome+"\nVRAM: "+gb+" GB\nTipo de VRAM: "+tipo+"\nConexão: "+conn+"\nHybrid Crossfire ("+igpu+"): "+valor+"\n";
 }
 // ↓ Mesma situação para manter atributos como private
 public String veriGPU() {return igpu;}
 public boolean verHybridCrossfire() {return hcf;}
 public void HybridCrossfire() {
  this.hcf = !this.hcf;
  this.valor = (hcf) ? "Ativado" : "Desativado";
  System.out.println(this.hcf ? "O Hybrid Crossfire está ativado." : "O Hybrid Crossfire está desativado.");
 }
}

File: components/hcfGPU.py
from .GPU import GPU # ← Esta classe é herdada da GPU, para placas de vídeo Radeon com Hybrid Crossfire
class hcfGPU(GPU):
 # ↓ Argumentos para criar a classe
 def __init__(self, nome:str, gb:int, tipo:str, conn:str, igpu:str):
  # ↓ Puxará atributos da classe mãe GPU
  super().__init__(nome, gb, tipo, conn)
  if igpu is None or igpu.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  
  self.__igpu=igpu # ← Placa de vídeo integrada compatível para Hybrid Crossfire
  self.__hcf=False # ← Tecnologia desligada por padrão
  self.__valor = "Ativado" if self.__hcf else "Desativado"
  
 # ↓ Simplesmente digitando o nome do instance no console, ou dentro do parênteses no "print()" irá renderizar a string abaixo
 def __repr__(self) -> str:
  return f"\n::::: TECHPOWERUP GPU DATABASE :::::\nNome: {self._nome}\nVRAM: {self._gb} GB\nTipo de VRAM: {self._tipo}\nConexão: {self._conn}\nHybrid Crossfire ({self.__igpu}): {self.__valor}\n"
 
 # ↓ Mesma situação para manter atributos como privados
 def veriGPU(self) -> str:
  return self.__igpu
 def verHybridCrossfire(self) -> bool:
  return self.__hcf
 def HybridCrossfire(self):
  self.__hcf = not(self.__hcf)
  self.__valor = "Ativado" if self.__hcf else "Desativado"
  print("O Hybrid Crossfire está ativado.") if self.__hcf else print("O Hybrid Crossfire está desativado.")
